### PR TITLE
feat: adding enterprise group checks in can_redeem

### PR DESCRIPTION
**Pull request by kiram15**
Updating the can_redeem function with an additional check to ensure that if the enterprise has groups, the learner is a member of one of those groups. 

[Jira Ticket ](https://2u-internal.atlassian.net/browse/ENT-8417)

### CODE REVIEW

File: enterprise_access/apps/subsidy_access_policy/models.py [hunk 1]
```diff
@@ -591,10 +592,29 @@ def can_redeem(self, lms_user_id, content_key, skip_customer_user_check=False):
 
         # learner not associated to enterprise
         if not skip_customer_user_check:
-            if self.lms_api_client.get_enterprise_user(self.enterprise_customer_uuid, lms_user_id) is None:
```
Comment (alex-sheehan-edx) [inline enterprise_access/apps/subsidy_access_policy/models.py:594]: I'm surprised this is None and not False 🤔 🤷

File: enterprise_access/apps/subsidy_access_policy/models.py [hunk 2]
```diff
@@ -591,10 +592,29 @@ def can_redeem(self, lms_user_id, content_key, skip_customer_user_check=False):
...
+                [str(group["uuid"]) for group in learner_record['user']['enterprise_group']]
+            )
+
+            policy_groups = PolicyGroupAssociation.objects.filter(subsidy_access_policy=self).all()
+            policy_groups_uuids = set(  # pylint: disable=consider-using-set-comprehension
+                [str(group.enterprise_group_uuid) for group in policy_groups]
+            )
+
+            # if there are groups associated with the policy, if not we want to be backwards compatible with
+            # enterprises who haven't implemented groups yet
```
Comment (alex-sheehan-edx) [inline enterprise_access/apps/subsidy_access_policy/models.py:610]: [tiny nit] technically it's policies that haven't implemented groups 😬

File: enterprise_access/apps/subsidy_access_policy/tests/test_models.py [hunk 3]
```diff
@@ -524,6 +556,13 @@ def test_learner_spend_cap_policy_can_redeem(
         if policy_is_active:
             policy_record = self.per_learner_spend_policy
 
+        PolicyGroupAssociationFactory(
+            enterprise_group_uuid=TEST_ENTERPRISE_GROUP_UUID,
+            subsidy_access_policy=policy_record
+        )
+
+        # import pdb;
```
Comment (alex-sheehan-edx) [inline enterprise_access/apps/subsidy_access_policy/tests/test_models.py:564]: NIT: cruft

**Review by alex-sheehan-edx — APPROVED** [review]
Tiny nits, looks good otherwise, reminder for the both of us to update logic related to the learner groups response payload if/when the change happens on the backend

File: enterprise_access/apps/subsidy_access_policy/models.py (hunk 1 above)
Comment (kiram15) [reply enterprise_access/apps/subsidy_access_policy/models.py:594]: This got changed with Katrina's earlier changes!

**Pull request merged**